gax/src/main/java/com/google/api/gax/core/GaxProperties.java

```diff
@@ -57,12 +57,16 @@ public static String getLibraryVersion(Class<?> libraryClass) {
    * method are expected to be cached.
    */
   public static String getLibraryVersion(Class<?> libraryClass, String propertyName) {
-    String version = getLibraryVersion(libraryClass);
-    if (!DEFAULT_VERSION.equals(version)) {
-      return version;
+    String version = null;
+    // Always read GaxProperties' version from the properties file.
+    if (!libraryClass.equals(GaxProperties.class)) {
+      version = getLibraryVersion(libraryClass);
+      if (!DEFAULT_VERSION.equals(version)) {
```

Comment:
why the if? seems we can return unconditionally here if it is the default version. Or do you want to read from gradle.properties instead if this is the default version?

Reply:
Since this seems applicable to any generic library class (i.e. keep reading if the version is empty, aka `DEFAULT_VERSION`), I was trying to keep the non-GaxProperties logic the same.